WordPress/gutenberg
# Columns block: reinstate tablet 2 col stacking

Related to https://github.com/WordPress/gutenberg/issues/40952

## What?
This is an experiment to reinstate the tablet breakpoint CSS for the columns block from https://github.com/WordPress/gutenberg/pull/37436 which forces two columns rather than a total stack.

## Why?
We removed the style as part of https://github.com/WordPress/gutenberg/pull/37436 , favouring to stack all column blocks below the desktop breakpoint, however it caused a visual regression.

## How?
1. Reinstating the `flex-basis` rule, ensuring we offset any current block gap values.
2. Restricting this behaviour to a new block style **"Tablet View"** (name debatable)

<img width="937" alt="Screen Shot 2022-05-31 at 10 47 03 am" src="https://user-images.githubusercontent.com/6458278/171072091-eb637efb-79b7-4f91-bd59-e08e9059e2cf.png">

## Testing Instructions

Check in classic and block themes that the columns please!

Insert a columns block with more than 1 column. Ensure that 'stack on mobile' is turned **on** to allow stacking.

<img width="278" alt="Screen Shot 2022-05-18 at 12 06 36 pm" src="https://user-images.githubusercontent.com/6458278/168942390-43eab099-2e6f-467b-ab3c-a92fe99f5ce2.png">

With the **"Tablet View"** applied to the columns block, and between  600px and 781px (the "tablet" view) the columns should stack 2 per row, with any trailing columns stretching the full width.

Turn **off** 'stack on mobile' and check that the columns no longer stack.

The editor and frontend should behave similarly. 

**Caveat**: adding a custom blockGap value that is greater than `var(--wp--style--block-gap, 2em)` to the columns block will trigger stacking since the spacing is now too great for a 2 per row stacking.


<details>

<summary>Test block code</summary>

```html

<div class="wp-block-columns">
<div class="wp-block-column has-background" style="background-color:#eaed48">
<p>Column</p>
</div>



<div class="wp-block-column has-background" style="background-color:#eaed48">
<p>Column</p>
</div>
</div>



<div class="wp-block-columns">
<div class="wp-block-column has-background" style="background-color:#ecffae">
<div class="wp-block-group">
<p>A paragraph inside a Group block inside a Column.</p>



<p>A paragraph inside a Group block inside a Column.</p>



<p>A paragraph inside a Group block inside a Column.</p>



<p>A paragraph inside a Group block inside a Column.</p>
</div>



<ul class="wp-block-social-links">























</ul>
</div>
</div>



<div class="wp-block-columns">
<div class="wp-block-column has-background" style="background-color:#ffccd8">
<p>Column</p>
</div>



<div class="wp-block-column has-background" style="background-color:#ffccd8">
<p>Column</p>
</div>



<div class="wp-block-column has-background" style="background-color:#ffccd8">
<p>Column</p>
</div>
</div>



<div class="wp-block-columns">
<div class="wp-block-column has-background" style="background-color:#b3002a">
<p>Column</p>
</div>



<div class="wp-block-column has-background" style="background-color:#b3002a">
<p>Column</p>
</div>



<div class="wp-block-column has-background" style="background-color:#b3002a">
<p>Column</p>
</div>



<div class="wp-block-column has-background" style="background-color:#b3002a">
<p>Column</p>
</div>



<div class="wp-block-column has-background" style="background-color:#b3002a">
<p>Column</p>
</div>
</div>


```

</details>

## Screenshots or screencast 

![2022-05-18 10 51 44](https://user-images.githubusercontent.com/6458278/169199939-174b9f10-2055-4145-9346-5ffc513df82c.gif)
